documentation/docs/20-core-concepts/50-state-management.md

```diff
@@ -19,41 +51,56 @@ declare module '$lib/user' {
 	const response = await fetch('/api/user');
+
+	// NEVER DO THIS!
 	user.set(await response.json());
 }
 ```
 
-If you are using SSR, the `load` function will run on the server initially, which means that the whole server instance which serves _all_ requests from _all_ users has its `user` state set to the one just requested from the API. To scope this to a single user, you have a couple of options:
+As with the previous example, this puts one user's information in a place that is shared by _all_ users. Instead, just return the data...
 
-- if you need to access the state only inside server `load` functions, use [`locals`](hooks#server-hooks-handle)
-- if you need to persist the state across reloads, but only need to access it inside `load` functions, use [`cookies` in server `load` functions](load#cookies-and-headers). If the state is more complex, safe a key to the state in the cookie to look it up in for example a database
-- if you need to access and update the state inside components, use Svelte's [context feature](https://svelte.dev/docs#run-time-svelte-setcontext). That way, the state is scoped to components, which means they are not shared across different requests on the server. The drawback is that you can only access the context at component initialization, which may make interacting with the store value a little trickier if you want to do that outside of components. SvelteKit's stores from `$app/stores` for example are setup like this (which is why you may have encountered a related error message)
+```diff
+/// file: +page.js
+export async function load({ fetch }) {
+	const response = await fetch('/api/user');
+
++	return {
++		user: await response.json()
++	};
+}
+```
+
+...and pass it around to the components that need it, or use [`$page.data`](/docs/load#$page-data).
+
+## Using stores with context
+
+You might wonder how we're able to use `$page.data` and other [app stores](/docs/modules#$app-stores) if we can't use our own stores. The answer is that app stores on the server use Svelte's [context API](https://learn.svelte.dev/tutorial/context-api) — the store is attached to the component tree with `setContext`, and when you subscribe you retrieve it with `getContext`. We can do the same thing with our own stores:
 
 ```svelte
-/// +layout.svelte
+/// file: +layout.svelte
 <script>
 	import { setContext } from 'svelte';
+	import { writable } from 'svelte/store';
 
 	/** @type {import('./$types').LayoutData} */
 	export let data;
 
-	// Create a store...
-	const user = writable(data.user);
-	// ...add it to the context for child components to access
+	// Create a store and update it when necessary...
+	const user = writable();
+	$: user.set(data.user);
+
+	// ...and add it to the context for child components to access
 	setContext('user', user);
-	// Optionally update the data everytime the load function is rerun
-	$: $user = data.user;
 </script>
 ```
 
 ```svelte
-/// +src/user/+page.svelte
+/// file: +src/user/+page.svelte
```

Comment:
```suggestion
/// file: user/+page.svelte
```
(or src/routes/user/+page.svelte, but then we also need to update the layout path above for consistency)